apps/web/src/components/workflow/preview/email/EmailPreview.tsx

```diff
@@ -1,207 +1,125 @@
-import { Grid, JsonInput, useMantineTheme } from '@mantine/core';
-import { useCallback, useEffect, useMemo, useState } from 'react';
-import { useFormContext, useWatch } from 'react-hook-form';
-
+import { JsonInput } from '@mantine/core';
+import { useEffect, useMemo, useState } from 'react';
+import { useFormContext } from 'react-hook-form';
+import styled from '@emotion/styled';
 import { Button, colors, inputStyles } from '@novu/design-system';
 
-import { IS_DOCKER_HOSTED, useDataRef } from '@novu/shared-web';
-import { useGetLocalesFromContent, usePreviewEmail } from '../../../../api/hooks';
-import { useActiveIntegrations, useAuthController, useProcessVariables } from '../../../../hooks';
+import { useActiveIntegrations, useProcessVariables } from '../../../../hooks';
 import type { IForm } from '../../../../pages/templates/components/formTypes';
 import { useStepFormErrors } from '../../../../pages/templates/hooks/useStepFormErrors';
 import { useStepFormPath } from '../../../../pages/templates/hooks/useStepFormPath';
 import { When } from '../../../utils/When';
 import { PreviewMobile } from './PreviewMobile';
 import { PreviewWeb } from './PreviewWeb';
+import { useTemplateLocales } from '../../../../pages/templates/hooks/useTemplateLocales';
+import { usePreviewEmailTemplate } from '../../../../pages/templates/hooks/usePreviewEmailTemplate';
+
+const PreviewContainer = styled.div`
+  display: flex;
+  flex-wrap: nowrap;
+  height: 100%;
+  margin: 0;
+  margin-bottom: 1.5rem;
+  gap: 1.5rem;
+`;
+
+const JSONContainer = styled.div`
+  min-width: 28rem;
+  height: 100%;
+  background: ${({ theme }) => (theme.colorScheme === 'dark' ? colors.B17 : colors.B98)};
+  border-radius: 7;
+`;
 
 export const EmailPreview = ({ showVariables = true, view }: { view: string; showVariables?: boolean }) => {
-  const { control } = useFormContext<IForm>();
+  const { watch } = useFormContext<IForm>();
   const path = useStepFormPath();
   const error = useStepFormErrors();
 
-  const subject = useWatch({
-    name: `${path}.template.subject`,
-    control,
-  });
-  const contentType = useWatch({
-    name: `${path}.template.contentType`,
-    control,
-  });
-  const htmlContent = useWatch({
-    name: `${path}.template.htmlContent`,
-    control,
-  });
-  const editorContent = useWatch({
-    name: `${path}.template.content`,
-    control,
-  });
-  const variables = useWatch({
-    name: `${path}.template.variables`,
-    control,
-  });
-  const layoutId = useWatch({
-    name: `${path}.template.layoutId`,
-    control,
-  });
-
-  const { integrations = [] } = useActiveIntegrations();
+  const contentType = watch(`${path}.template.contentType`);
+  const htmlContent = watch(`${path}.template.htmlContent`);
+  const editorContent = watch(`${path}.template.content`);
+  const variables = watch(`${path}.template.variables`);
```

Comment:
Use watch instead of `useWatch` because the `path` is URL based and when it's changing we want to retrieve the appropriate values from the form context. `useWatch` is subscribing to `path` only once.